discussion_forum/api.py

```diff
@@ -0,0 +1,500 @@
+            HTTP_401_UNAUTHORIZED: Raised by `IsInstructorOrTAOrStudent`
+                                   permission class
+            HTTP_403_FORBIDDEN: Raised by `IsInstructorOrTAOrStudent` permission class
+        """
+        user = request.user
+        discussion_forum = request.data["discussion_forum"]
+        course_id = DiscussionForum.objects.get(id=discussion_forum).course.id
+        check = self._is_registered(course_id, user)
+        if check is not True:
+            return check
+
+        serializer = self.get_serializer(data=request.data)
+
+        if serializer.is_valid():
+            serializer.save()
+            return Response(serializer.data, status=status.HTTP_201_CREATED)
+        errors = serializer.errors
+        logger.error(errors)
+        return Response(errors, status=status.HTTP_400_BAD_REQUEST)
+
+    @action(detail=True, methods=["GET"])
+    def list_discussion_threads(self, request, pk):
+        """Gets all the discussion threads in the discussion forum
+           with primary key as pk.
+
+        Args:
+            request (Request): DRF `Request` object
+            pk (int): Primary key of the discussion_forum
+
+        Returns:
+            `Response` with all the discussion threads data
+             and status HTTP_200_OK
+
+        Raises:
+            HTTP_404_NOT_FOUND: Raised by `_is_registered()` method
+            HTTP_401_UNAUTHORIZED: Raised by `IsInstructorOrTAOrStudent`
+                                              permission class
+            HTTP_403_FORBIDDEN: Raised by `IsInstructorOrTAOrStudent` permission class
+        """
+        check = self._is_registered(pk, request.user)
+        if check is not True:
+            return check
+
+        discussion_threads = DiscussionThread.objects.filter(discussion_forum_id=pk)
+        page = self.paginate_queryset(discussion_threads)
+        if page is not None:
+            serializer = self.get_serializer(page, many=True)
+            return self.get_paginated_response(serializer.data)
+
+        serializer = self.get_serializer(discussion_threads, many=True)
+        return Response(serializer.data)
+
+    @action(detail=True, methods=["GET"])
+    def retrieve_discussion_thread(self, request, pk):
+        """Gets the discussion_thread with primary key as pk.
+
+        Args:
+            request (Request): DRF `Request` object
+            pk (int): Primary key of the discussion_thread
+
+        Returns:
+            `Response` with the discussion_thread data and status HTTP_200_OK
+
+        Raises:
+            HTTP_404_NOT_FOUND: Raised by `_is_registered()` method
+            HTTP_401_UNAUTHORIZED: Raised by `IsInstructorOrTAOrStudent`
+                                   permission class
+            HTTP_403_FORBIDDEN: Raised by `IsInstructorOrTAOrStudent` permission class
+        """
+        discussion_thread = self.get_object()
+        discussion_forum_id = discussion_thread.discussion_forum.id
```

Comment:
line 130-139 is not required as self.get_object() handles that

Reply:
This is querying discussion forum table. get_object must be taking care of discussion thread table.